### ISSUE TITLE
Add VeraCrypt and alternative encryption support

### ISSUE THREAD

**Issue by GitGangGuy** [QubesOS/qubes-issues#5875]
**The problem you're addressing (if any)**
As it currently stands cascade ciphers, advanced hashing algorithms, advanced randomness generation and more can only be found in more sophisticated approaches like VeraCrypt. However, Qubes doesn't allow for additional flags when opening an encrypted volume/partition, leaving only LUKS with the limited cryptsetup options  for disk encryption.

**Describe the solution you'd like**
cryptsetup is awesome with its defaults, and its more than enough for most people. However, users like me would like to have more control over what (full) disk encryption we use. In my case, I would like to use VeraCrypt, but I'm having trouble using it. I followed the custom installation guide, but used veracrypt beforehand (since I can't install veracrypt from the installer terminal) to encrypt my hard drive. I'm not sure what to do next; currently, QubesOS doesn't provide an option to pass additional flags to cryptsetup on boot (`--veracrypt` + `--veracrypt-query-pim` for rudimentary veracrypt support) or use a different command-line decryption tool all together.

**Where is the value to a user, and who might that user be?**
I believe it is fundamentally important to have the user decide if they want to use LUKS or VeraCrypt or even TrueCrypt or BitLocker, especially with a distribution whose primary focus is security. Every single user out there might have different expectations for their disk encryption; some don't care to encrypt the boot partition; some want performance over security; others security over performance; some want something inbetween. QubesOS is currently forcing its expectations onto its users, which is why I'd really like to see this moving forward.

**Describe alternatives you've considered**
I wasn't able to use my veracrypt disk in QubesOS. I'm not sure if I've missed something, but I'd also like to note that besides the very scarce "Custom Installation" documentation page, there's no information on this.

**Relevant [documentation](https://www.qubes-os.org/doc/) you've consulted**

- https://www.qubes-os.org/doc/custom-install/

I really love open source, and I love this OS (which is why I want to use it over something like Arch). I also know that maintaining such a project is extremely difficult and pressurizing, and I don't want to add onto the pile of probably more important stuff to do. Yet, I hope I'll at some point be able to use the encryption I want to use. Definitely respond here if I missed something or so. Keep up the great work!

**Comment by andrewdavidwong** [comment]
I think the short answer here is:

 - We mostly borrow the installer from upstream Fedora.
 - It sounds like the feature you're requesting would probably have to be a Fedora feature request.
 - Our Custom Installation page is just about customizing the LUKS config.
 - Security-wise, anything beyond customizing the LUKS config is overkill, and even that might be overkill. You're much more likely to face a security shortfall somewhere besides the FDE.

Bottom line: If this were a Qubes issue, it would probably be extremely low priority, only done with a community contribution, but I think it's probably more of a Fedora request than one for Qubes.

For these reasons, I'm going to close this issue as "not our bug" (technically, "not our feature request"). If anyone believes that this is a mistake or believes that this should be reopened, please leave a comment letting us know.

**Comment by straussmaus** [comment]
@andrewdavidwong 

I think maybe the request is for Veracrypt to be a default package installed with dom0?  I don't see a mention of Fedora anywhere?  I don't know if that changes the priority/interest.  I am just commenting on what I notice.

**Comment by andrewdavidwong** [comment]
> I think maybe the request is for Veracrypt to be a default package installed with dom0?

In that case, the answer would most likely still be no, for the reasons explained here:

https://www.qubes-os.org/faq/#could-you-please-make-my-preference-the-default

https://www.qubes-os.org/faq/#what-is-qubes-attitude-toward-changing-guest-distros

For the reasons already hashed out in this issue, it's not something that's important for all users. It's mainly something that this particular user wants for themselves. In that case, building one's own Qubes ISO is an option:

https://www.qubes-os.org/doc/qubes-iso-building/


> I don't see a mention of Fedora anywhere?

Fedora is the distro used in dom0 at present, hence also the installer. See my comment above.
